arctic/store/version_store.py

```diff
@@ -526,7 +526,7 @@ def _insert_version(self, version):
             # upon intermittent Mongo errors
             # If, however, we get a DuplicateKeyError, suppress it and raise OperationFailure, so that the method-scoped
             # mongo_retry re-tries and creates a new version, to overcome the issue.
-            mongo_retry(self._versions.insert_one)(version)
+            mongo_retry(self._versions.insert)(version, check_keys=False)
```

Comment:
`insert_one()` doesn't support `check_keys`

Reply:
Unfortunately I suspect the `check_keys` functionality is deprecated along with `insert`, as it doesn't appear in any of the new methods.

Reply:
Good point. Reading around it seems that while mongo supports dots in field names, they are strongly discouraged. Given it's not supported in the new API (and I can't find any hints that it will ever be), this change could come back to bite us.